cmake/SeracBasics.cmake

```diff
@@ -43,7 +43,9 @@ endif()
 
 option(SERAC_ENABLE_PROFILING "Enable profiling functionality" OFF)
 
-cmake_dependent_option(SERAC_ENABLE_BENCHMARKS "Enable benchmark executables" ON "ENABLE_BENCHMARKS" OFF)
+if (ENABLE_BENCHMARKS)
+    set(SERAC_ENABLE_BENCHMARKS ON)
+endif()
```

Comment:
While `cmake_dependent_option` is admittedly confusing I think it is okay as is. `SERAC_ENABLE_BENCHMARKS` generally shouldn't be set explicitly unless you want to enable benchmarks yet disable Serac's. Simply set `ENABLE_BENCHMARKS`.

`SERAC_ENABLE_BENCHMARKS` can only be set if `ENABLE_BENCHMARKS` is `ON`. If `ENABLE_BENCHMARKS` is `ON`, it will automatically set `SERAC_ENABLE_BENCHMARKS` to `ON`. Otherwise `SERAC_ENABLE_BENCHMARKS` is forced to `OFF`.

Reply:
please re-add `cmake_dependent_option(SERAC_ENABLE_BENCHMARKS "Enable benchmark executables" ON "ENABLE_BENCHMARKS" OFF)`